far/stddlg.cpp

```diff
@@ -190,17 +192,10 @@ SearchReplaceDlgResult GetSearchReplaceString(
 
 	const auto SetFlagIf{ [&](const item_id Item, const auto Flag, const bool Condition) { if (Condition) DlgItems[Item].Flags |= Flag; } };
 	const auto SetSelected{ [&](const item_id Item, const bool Selected) { DlgItems[Item].Selected = Selected; } };
-	const auto SetStringIf{ [&](const item_id Item, const string_view String, const bool Condition) { if (Condition) DlgItems[Item].strData = String; } };
+	const auto SetStringIf{ [&](const item_id Item, auto&& String, const bool Condition) { if (Condition) DlgItems[Item].strData = std::move(std::forward<decltype(String)>(String)()); }};
```

Comment:
The idea was to avoid expensive `BlobToHexString` call for `dlg_edit_search_hex` if the caller has no Hex option. If you think the optimization is not worth complexity, let me know, I'll revert it.

Reply:
> The idea was to avoid expensive `BlobToHexString` call

Ah, I misread it - `String` is actually a callable!
The optimization makes sense, but:
- Please rename `auto&& String` to make it more obvious that it's a callable rather than an actual string, e.g. `StringGetter` or something like that.
- `std::move(std::forward<decltype(String)>(;` is redundant, `String()` will do the same thing.
- `auto&&` can be replaced with `function_ref<string()> const` to avoid templatizing.

Reply:
Renamed and removed `move-forward` nonsense (do not know what happened).

`function_ref<string()>` involves type erasure and indirection, hence most likely actual call, while `auto&&` should be nicely inlined. Let's keep it this way.